File: Data Structures & Algorithms/max-water-container/submission-2.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let left  = 0; 
        let right = heights.length-1;
        let areamax =0;

        while(left<right){
        if(heights[left]>heights[right]){
            areamax= Math.max(areamax,heights[right]*(right-left));
            right--;
        }
        else{
           areamax= Math.max(areamax,heights[left]*(right-left));
            left++;
        }
    }
     return areamax;
            
        }
       

    }    

File: Data Structures & Algorithms/maximum-product-subarray/submission-2.java
class Solution {
    public int maxProduct(int[] nums) {
        int product= 1; 
        int ans = nums[0];

        for(int i =0 ; i<nums.length ; i++){

            product = product==0 ? 1:product;

            product *= nums[i];

            ans=Math.max(ans,product) ;
        }

        return ans;
        
    }
}
